addons/mass_mailing/wizard/mailing_mailing_test.py

```diff
@@ -23,9 +23,72 @@ def _default_email_to(self):
             ('create_uid', '=', self.env.uid),
         ], order='create_date desc', limit=1).email_to or self.env.user.email_formatted
 
+    def _get_model_selection(self):
+        result = self.env["ir.model"].sudo().search([("transient", "=", False)])
+        return [(model.model, model.name) for model in result]
+
     email_to = fields.Text(string='Recipients', required=True,
                            help='Carriage-return-separated list of email addresses.', default=_default_email_to)
     mass_mailing_id = fields.Many2one('mailing.mailing', string='Mailing', required=True, ondelete='cascade')
+    email_from = fields.Char(string='From', related='mass_mailing_id.email_from')
+    subject = fields.Char(string='Subject', compute='_compute_subject')
+    body_html = fields.Html(string='Preview Body', compute='_compute_body_preview', sanitize='email_outgoing')
+    preview_record_ref = fields.Reference(string='Preview for', selection=_get_model_selection, compute='_compute_test_recipient_ref')
+    is_mobile = fields.Boolean(store=False)
+
+    @api.depends('preview_record_ref')
+    def _compute_subject(self):
+        for record in self:
+            mailing = record.mass_mailing_id
+            if not record.preview_record_ref:
+                record.subject = mailing.subject or ""
+                continue
+            record.subject = mailing._render_field(
+                'subject', record.preview_record_ref.ids, compute_lang=False
+            )[record.preview_record_ref.id]
+
+    @api.depends('mass_mailing_id.mailing_model_real')
+    def _compute_test_recipient_ref(self):
+        for record in self:
+            model = record.mass_mailing_id.mailing_model_real
+            res = self.env[model].search([], limit=1)
+            record.preview_record_ref = f'{model},{res.id}' if res else False
+
+    @api.depends('preview_record_ref')
+    def _compute_body_preview(self):
+        with file_open('mass_mailing/static/src/scss/mass_mailing_mail.scss', 'r') as file:
+            styles = file.read()
+        for record in self:
+            mailing = record.mass_mailing_id
+            if not record.preview_record_ref:
+                record.body_html = mailing.body_html
+                continue
+
+            body_html = mailing._render_field(
+                'body_html',
+                record.preview_record_ref.ids,
+                compute_lang=False,
+                options={'preserve_comments': True},
+            )[record.preview_record_ref.id]
+            record.body_html = self.env['ir.qweb']._render(
+                'mass_mailing.mass_mailing_mail_layout',
+                {
+                    'body': body_html,
+                    'mailing_style': Markup('<style>%s</style>') % styles,
+                },
+                minimal_qcontext=True,
+            )
+
+    def action_open_test_email(self):
```

Comment:
A method on test wizard allowing to open the form view of the same test wizard ? Makes no sense, to simplify.